crypto/sm2/sm2_pmeth.c

```diff
@@ -63,6 +66,13 @@ static int pkey_sm2_copy(EVP_PKEY_CTX *dst, EVP_PKEY_CTX *src)
             return 0;
         }
     }
+
+    dctx->uid = OPENSSL_strdup(sctx->uid);
+    if (sctx->uid != NULL && dctx->uid == NULL) {
```

Comment:
Even though OPENSSL_strdup can handle NULL, I'd still argue in favour of same semantic as conventional strdup. I.e. check sctx->uid for NULL *first*. It's more readable that way. Or in other words for example
```
if (sctx->uid != NULL
    && (dctx->uid = OPENSSL_strdup(sctx->uid) == NULL)
```